File: 2022/day_7/day_7.js
fs = require("fs");

let directory = {};
let currentDirectory = '/';

fs.readFile("input.txt", "utf8", (err, data) => {
  if (err) throw err;
  const lines = data.split("\n");
  lines.forEach((line) => {
    lineArr = line.split(" ");
    if (lineArr[0] == '$') {
        // instructions
        const instruction = lineArr[1];

        if(instruction == 'cd') {
            const pathTo = lineArr[2];
            if(pathTo === '/'){ 
                currentDirectory = directory;
            } else {
                if(directory[pathTo] ) {
                    currentDirectory = 
                }
                
            }
        }
    }
  });
});

File: 2022/day_7/day_7.py
MAX_DIRECTORY_SIZE = 100000

input = ''

with open('input.txt') as file:
    input = file

for line_input in input:
    line = line_input.strip().split(' ')

    # if current line is an instruction
    if line[0] == '$':
        if line[1] == 'cd':
            
    else: